application/controllers/UploaderController.php

```diff
@@ -143,11 +143,16 @@ public function run($actionID)
             $filename = sanitize_filename($_FILES['uploadfile']['name'], false, false, true);
             $size = $_FILES['uploadfile']['size'] / 1024;
             $preview = Yii::app()->session['preview'];
-            $aFieldMap = createFieldMap($oSurvey, 'short', false, false, $sLanguage);
-            if (!isset($aFieldMap[$sFieldName])) {
-                throw new CHttpException(400); // See for debug > 1
+            /* Find the question by sFieldName : must be a upload question type, and id is end of sFieldName in $surveyid*/
+            $aFieldName = explode("X", $sFieldName);
+            if (empty($aFieldName[2]) || !ctype_digit($aFieldName[2])) {
+                throw new CHttpException(400);
+            }
+            $oQuestion = Question::model()->find("sid = :sid and qid = :qid and type = :type", [":sid" => $surveyid, ":qid" => $aFieldName[2], ":type" =>  Question::QT_VERTICAL_FILE_UPLOAD]);
```

Comment:
Since we do this twice (even though not in the same use-case), it would be feasible to have a method for this purpose and call it from both locations.

Reply:
Yep, but in my opinion UploaderControler need more rewrite ;) if elseif elseif elseif …

private method inside UploaderController or a Quetion method ?

Reply:
We agree that it needs more rewrite, but we cannot do a full rewrite as part of this work, it was not approved. If we manage to get approval for a full rewrite, we will need to create a separate ticket for that.

> private method inside UploaderController or a Quetion method ?

I'm good with either way